dubbo-registry/dubbo-registry-kubernetes/src/main/java/org/apache/dubbo/registry/kubernetes/KubernetesServiceDiscovery.java

```diff
@@ -194,21 +194,22 @@ public ServiceInstance getLocalInstance() {
 
     @Override
     public void addServiceInstancesChangedListener(ServiceInstancesChangedListener listener) throws NullPointerException, IllegalArgumentException {
+        super.addServiceInstancesChangedListener(listener);
         listener.getServiceNames().forEach(serviceName -> {
             SERVICE_UPDATE_TIME.put(serviceName, new AtomicLong(0L));
 
             // Watch Service Endpoint Modification
-            watchEndpoints(listener, serviceName);
+            watchEndpoints(serviceName);
```

Comment:
Does this change still match the original meaning?

The original impl only notify one listener while the new one notifies all listeners.